### PR TITLE
Seen: show when the target is currently in the channel

### PR DESCRIPTION
**Pull request by matiasw**
Adds functionality to report on if the user is currently in the channel.

### CODE REVIEW

File: plugins/Seen/plugin.py [hunk 1]
```diff
@@ -277,8 +289,14 @@ def _last(self, irc, channel, any=False):
             db = self.db
         try:
             (when, said) = db.seen(channel, '<last>')
-            reply = format(_('Someone was last seen in %s %s ago'),
-                             channel, utils.timeElapsed(time.time()-when))
+            pattern = r'<([^>]*)>'
```
Comment (jlu5) [inline plugins/Seen/plugin.py:292]: `r'<(.*?)>'` is easier to read

File: plugins/Seen/plugin.py [hunk 2]
```diff
@@ -277,8 +289,14 @@ def _last(self, irc, channel, any=False):
...
+            try:
+                nick = re.search(pattern, str(said)).group(1)
+            except AttributeError:
+                irc.error(format(_('I couldn\'t parse the nick of the speaker of the last line.')))
```
Comment (jlu5) [inline plugins/Seen/plugin.py:296]: I would prefer explicitly checking for whether the match exists:

```python
match = re.search(pattern, said)
if not match:
    irc.error(..., Raise=True)  # Raise=True implicitly returns
nick = match.group(1)
```

File: plugins/Seen/test.py [hunk 3]
```diff
@@ -85,10 +85,20 @@ def testSeen(self):
...
         self.assertError('seen *')
         self.assertNotError('seen %s' % self.nick)
+        # Test case: 'seen' with a nick (user in channel)
+        self.assertRegexp('seen %s' %self.nick, 'is in the channel now')
         m = self.assertNotError('seen %s' % self.nick.upper())
         self.assertIn(self.nick.upper(), m.args[1])
         self.assertRegexp('seen user %s' % self.nick,
                           '^%s was last seen' % self.nick)
+        # Test case: 'seen' with a user (user in channel)
+        self.assertRegexp('seen %s' %self.nick, 'is in the channel now')
```
[the remark this answers is not in the document]
Comment (matiasw) [reply plugins/Seen/test.py]: Yes; sorry.

File: plugins/Seen/test.py [hunk 4]
```diff
@@ -85,10 +85,20 @@ def testSeen(self):
         self.assertNotError('config plugins.Seen.minimumNonWildcard 2')
         self.assertError('seen *')
         self.assertNotError('seen %s' % self.nick)
+        # Test case: 'seen' with a nick (user in channel)
+        self.assertRegexp('seen %s' %self.nick, 'is in the channel now')
```
[the remark this answers is not in the document]
Comment (jlu5) [reply plugins/Seen/test.py:89]: Sorry, I meant to say there _should_ be a space around `%` on both sides. It's worth taking look at the Python style guide https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements which we mostly try to follow

[the remark this answers is not in the document]
Comment (matiasw) [reply plugins/Seen/test.py]: Maybe de89e81cdf2b1ef8ac410a77da47bebd4bfdbbd3 is more like it?

**Comment by progval** [comment]
Sorry for not mentioning it earlier, but could you split the tests? There should only be one case per function, it makes them easier to understand

**Comment by matiasw** [comment]
Maybe like in 0aec1ff5747e007ed746e0b168bed27dcc20c290 ?

**Comment by progval** [comment]
Thanks!

**Pull request merged**